specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/extensions.json

```diff
@@ -176,6 +176,148 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/extensions/azureMonitor": {
+      "put": {
+        "tags": [
+          "Extensions"
+        ],
+        "description": "Enables the Azure Monitor on the HDInsight cluster.",
+        "x-ms-examples": {
+          "Enable cluster monitoring": {
+            "$ref": "./examples/EnableLinuxClusterAzureMonitor.json"
+          }
+        },
+        "operationId": "Extensions_EnableAzureMonitor",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AzureMonitorRequest"
```

Comment:
The PUT request body doesn't match the ARM standard pattern at all. I see properties workspaceId, primaryKey are exposed at top-level instead of having them under properties envelope. For more details, refer RPC here https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope

However, I understand the swagger already defined the extensions API and this PR is just adding an instance of extension resource. I would give the exception for this reason and sign off.

I would recommend to have a workItem/task on your side to correct the request/response formats in future new API versions.

Reply:
Hi @RamyasreeChakka, Thanks for your comment and approval.

it seems that the we follow the rule:  the `workspaceId` and `primaryKey` under that properties of the AzureMonitorRequest definion.  Could you please help me understand better about this rule so that our service can satisfy the ARM standard pattern next time?

```
"AzureMonitorRequest": 
{      
  "description": "The azure monitor parameters.", 
    "properties": 
      {       
         "workspaceId":  {          
              "description": "The Log Analytics workspace ID.", 
                "type": "string"       
           },        
           "primaryKey": {
             "description": "The Log Analytics workspace key.",  
                 "type": "string"        
           },       
          "selectedConfigurations": {        
            "description": "The selected configurations.",   
             "$ref": "#/definitions/AzureMonitorSelectedConfigurations"       
            } 
      }
}
```